### ISSUE TITLE
:location is not automatically merged into response when :handle-see-other is defined 

### ISSUE THREAD

**Issue by jbarber** [clojure-liberator/liberator#198]
When dealing with a redirect-on-post pattern, I want to set information in the Ring session about the request as follows:
```clojure
((->
   (liberator.core/resource
     :allowed-methods [:post]
     :post-redirect? {:location "http://www.example.com/"}
     :handle-see-other (fn [ctx]
                         (liberator.representation/ring-response
                           {:flash ::submitted})))
   ring.middleware.params/wrap-params
   ring.middleware.session/wrap-session
   ring.middleware.flash/wrap-flash)
 (ring.mock.request/request :post "/any" {:email "foo@example.com"}))
```

However, the response does not include the Location header and requires that it be set by the user:
```clojure
     :handle-see-other (fn [ctx]
                         (liberator.representation/ring-response
                           {:flash ::submitted
                            :headers {"Location" (ctx :location)}})))
```

If :location is set in the context, it should be merged automatically if handle-see-other doesn't set it.

This should probably also occur for the other responses with codes of 300.

Additionally, as the Location field is a SHOULD for the 300's status code responses - should an exception be raised if it isn't present?

This problem was originally raise in the liberator group:
https://groups.google.com/forum/#!topic/clojure-liberator/X714xMpi4lY

**Comment by dyba** [comment]
:+1:

**Comment by ordnungswidrig** [comment]
This should be fixed in 0.13 by #214, (this issue was duplicated by #212 )
